src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeEventInfo.cs

```diff
@@ -68,6 +69,12 @@ public override string ToString()
 
             return m_addMethod.GetParametersNoCopy()[0].ParameterType.FormatTypeName() + " " + Name;
         }
+
+        public override bool Equals(object? obj) =>
+            obj == (object)this ||
+            (RuntimeTypeMetadataUpdateHandler.MetadataUpdaterSupportedAndCacheCleared && CacheEquals(obj));
+
+        public override int GetHashCode() => HashCode.Combine(m_token.GetHashCode(), RuntimeTypeHandle.GetModule(m_declaringType).GetHashCode());
```

Comment:
You can just combine the hashcode of the type and the token. You do not need to go to the module.

Reply:
Similarly, Equals can just compare the type.

Reply:
Can apply for all other members?

Reply:
I think so. Also, the cache equals method can use that too.